Sources/AsyncHTTPClient/ConnectionPool/HTTPConnectionPool.swift

```diff
@@ -18,89 +18,102 @@ enum HTTPConnectionPool {
     struct Connection: Hashable {
         typealias ID = Int
 
-        // PLEASE NOTE:
-        // The HTTP/1.1 connection code here is commented out, for a sad and simple reason: We
-        // don't have a HTTP1Connection yet. As soon as the HTTP1Connection has landed
-        // (https://github.com/swift-server/async-http-client/pull/400) we will enable
-        // HTTP1Connections here. Landing the connection box now enables us to already review the
-        // ConnectionPool StateMachines.
-
         private enum Reference {
-//            case http1_1(HTTP1Connection)
-
+            case http1_1(HTTP1Connection)
+            case http2(HTTP2Connection)
             case __testOnly_connection(ID, EventLoop)
         }
 
         private let _ref: Reference
 
-//        fileprivate static func http1_1(_ conn: HTTP1Connection) -> Self {
-//            Connection(_ref: .http1_1(conn))
-//        }
+        fileprivate static func http1_1(_ conn: HTTP1Connection) -> Self {
+            Connection(_ref: .http1_1(conn))
+        }
+
+        fileprivate static func http2(_ conn: HTTP2Connection) -> Self {
+            Connection(_ref: .http2(conn))
+        }
 
         static func __testOnly_connection(id: ID, eventLoop: EventLoop) -> Self {
             Connection(_ref: .__testOnly_connection(id, eventLoop))
         }
 
         var id: ID {
             switch self._ref {
-//            case .http1_1(let connection):
-//                return connection.id
+            case .http1_1(let connection):
+                return connection.id
+            case .http2(let connection):
+                return connection.id
             case .__testOnly_connection(let id, _):
                 return id
             }
         }
 
         var eventLoop: EventLoop {
             switch self._ref {
-//            case .http1_1(let connection):
-//                return connection.channel.eventLoop
+            case .http1_1(let connection):
+                return connection.channel.eventLoop
+            case .http2(let connection):
+                return connection.channel.eventLoop
             case .__testOnly_connection(_, let eventLoop):
                 return eventLoop
             }
         }
 
-        @discardableResult
-        fileprivate func close() -> EventLoopFuture<Void> {
+        fileprivate func execute(request: HTTPExecutableRequest) {
             switch self._ref {
-//            case .http1_1(let connection):
-//                return connection.close()
-
-            case .__testOnly_connection(_, let eventLoop):
-                return eventLoop.makeSucceededFuture(())
+            case .http1_1(let connection):
+                return connection.execute(request: request)
+            case .http2(let connection):
+                return connection.executeRequest(request)
```

Comment:
That's some mildly irritating asymmetry :)

Reply:
Fixed.